AnkiDroid/src/test/java/com/ichi2/anki/pages/PostRequestHandlerTest.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (c) 2024 voczi <dev@voczi.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify it under
+ *  the terms of the GNU General Public License as published by the Free Software
+ *  Foundation; either version 3 of the License, or (at your option) any later
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.anki.pages
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.ichi2.anki.RobolectricTest
+import com.ichi2.testutils.HamcrestUtils.containsInAnyOrder
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.InputStreamReader
+
+@RunWith(AndroidJUnit4::class)
+class PostRequestHandlerTest : RobolectricTest() {
+    @Test
+    fun `All backend typescript functions should be handled`() {
+        val assetManager = targetContext.assets
+        val tsFuncsStream = InputStreamReader(assetManager.open("backend/ts_funcs.txt"))
+        val tsFuncs = tsFuncsStream.readLines()
+        assertThat("List of stored Typescript functions is not empty", tsFuncs.isNotEmpty())
+
+        val collectionMethods = collectionMethods
+        val uiMethods = uiMethods
+        val combined = (collectionMethods + uiMethods).keys
+        assertThat(
+            "Mapping exists for every TS backend function call",
+            tsFuncs,
+            containsInAnyOrder(combined)
```

Comment:
If multiple people are confused, a clarifying comment is an easy way to eliminate confusion

My approval was contingent on clarification of what this assertion is doing and since I didn't trust it I proposed a test that probed failure if there were extra elements or missing elements, and I approved that.

That was reverted - which is fine, but only if my original confusion on what the extent of this assertion does is clarified for a Future Me where I've forgotten why this incompletely named assertion was adequate to protect against a future state where the ts funcs list has shrunk and we have stale bindings laying around, incurring maintenance cost for no reason

Clarifying comment suggestion that would satisfy my concern:

```suggestion
            // this matcher asserts equality in everything but order, no extras, nothing missing
            containsInAnyOrder(combined)
```

Reply:
I'd write an alias to this method in `AnkiAssert`, calling it `containsExactlyInAnyOrder` (or similar).

Therefore we won't need a comment, and the intention and functionality is more clear. We're working around `Hamcrest`'s naming/conventions here no matter what we do.